frontend/packages/dev-console/src/components/topology/list-view/TopologyListView.tsx

```diff
@@ -139,6 +172,91 @@ const TopologyListView: React.FC<TopologyListViewProps> = ({
     };
   }, [visualization, selectedId, applicationGroups, unassignedItems, onSelect]);
 
+  React.useEffect(() => {
+    let metricsInterval: any = null;
+    let monitoringAlertInterval: any = null;
+
+    const fetchMetrics = () => {
+      if (!PROMETHEUS_TENANCY_BASE_PATH) {
+        return;
+      }
+
+      const queries = {
+        memory: `sum(container_memory_working_set_bytes{namespace='${namespace}',container='',pod!=''}) BY (pod, namespace)`,
+        cpu: `pod:container_cpu_usage:sum{namespace="${namespace}"}`,
+      };
+
+      const promises = _.map(queries, (query, name) => {
+        const url = `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/query?namespace=${namespace}&query=${encodeURIComponent(
+          query,
+        )}`;
+        return coFetchJSON(url).then(({ data: { result } }) => {
+          const byPod: MetricValuesByPod = result.reduce((acc, { metric, value }) => {
+            acc[metric.pod] = Number(value[1]);
+            return acc;
+          }, {});
+          return { [name]: byPod };
+        });
+      });
+
+      Promise.all(promises)
+        .then((data) => {
+          const updatedMetrics = data.reduce(
+            (acc: OverviewMetrics, metric): OverviewMetrics => _.assign(acc, metric),
+            {},
+          );
+          updateMetrics(updatedMetrics);
+        })
+        .catch((res) => {
+          const status = _.get(res, 'response.status');
+          // eslint-disable-next-line no-console
+          console.error('Could not fetch metrics, status:', status);
+          // Don't retry on some status codes unless a previous request succeeded.
+          if (_.includes([401, 403, 502, 503], status) && _.isEmpty(metrics)) {
+            throw new Error(`Could not fetch metrics, status: ${status}`);
+          }
+        })
+        .then(() => {
+          metricsInterval = setTimeout(fetchMetrics, METRICS_POLL_INTERVAL);
+        })
+        .catch((e) => {
+          console.error(e); // eslint-disable-line no-console
+        });
```

Comment:
this logic is same as used in Admin side but that's a class based and this is functional connected component https://github.com/openshift/console/blob/master/frontend/public/components/overview/index.tsx#L267 , can we separate it and use across or it'll be beyond scope of this issue WDYT?

Reply:
I can look into doing this. Eventually we will be replacing the workloads tab contents with the topology list view and the duplication will be removed (see https://github.com/openshift/console/pull/6224)